### DISCUSSION TITLE
What caused the panic here?

### DISCUSSION THREAD

**Discussion by hiaro81** [bevyengine/bevy#6876] [Q&A] [answered]
What caused the panic here?

```
impl<T> Command for Insert<T>
where
    T: Bundle + 'static,
{
    fn write(self, world: &mut World) {
        if let Some(mut entity) = world.get_entity_mut(self.entity) {
            entity.insert(self.bundle);
        } else {
            panic!("error[B0003]: Could not insert a bundle (of type `{}`) for entity {:?} because it doesn't exist in this World.", std::any::type_name::<T>(), self.entity);
        }
    }
}
````

I am making an online game.
So my logic allows despawn entities when the client connection closed.

Is that panic a necessary code?

Error
thread 'main' panicked at 'error[B0003]: Could not insert a bundle (of type `geo::ecs::systems::external::ProtoPacketWrapper<sol_proto::geo::sync::MoveCmd>`) for entity 156v0 because it doesn't exist in this World.', /Users/kwangmoyang/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_ecs-0.9.1/src/system/commands/mod.rs:840:13
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: <bevy_ecs::system::commands::Insert<T> as bevy_ecs::system::commands::Command>::write
             at /Users/kwangmoyang/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_ecs-0.9.1/src/system/commands/mod.rs:840:13
   3: bevy_ecs::system::commands::command_queue::CommandQueue::push::write_command
             at /Users/kwangmoyang/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_ecs-0.9.1/src/system/commands/command_queue.rs:42:13
   4: bevy_ecs::system::commands::command_queue::CommandQueue::apply
             at /Users/kwangmoyang/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_ecs-0.9.1/src/system/commands/command_queue.rs:96:17

**Comment by MrGVSV** [comment, marked as answer]
You shouldn't be inserting components on despawned entities. You'll need to add a mechanism that removes the stored `Entity` (if you store it as `Option<Entity>` then just set it to `None`) when the client disconnects as it's now an invalid entity.

**Reply by hiaro81** [reply]
Okay, thank you. I think we need to separate our system in that way.
